src/Illuminate/View/Compilers/BladeCompiler.php

```diff
@@ -317,7 +318,7 @@ protected function compileComponentTags($value)
             return $value;
         }
 
-        return app(ComponentTagCompiler::class, [
+        return Container::getInstance()->make(ComponentTagCompiler::class, [
```

Comment:
This is a hack. You need to inject the component tag compiler instance. The compiler should not know about the container.

Reply:
I see. I was just looking at other examples and it seems like this is used in a few places, even in the same ComponentTagCompiler. I don't see how it can be injected without changing too much, so probably just going to instantiate my own object. Thanks for the feedback tho.